src/shammodels/ramses/include/shammodels/ramses/SolverConfig.hpp

```diff
@@ -76,6 +76,21 @@ namespace shammodels::basegodunov {
         }
     };
 
+    /**
+     * @brief Npscal_gas is the number of gas passive scalars
+     */
+    struct PassiveScalarGasConfig {
+        u32 npscal_gas = 0;
+
+        inline bool is_gas_passive_scalar_on() {
+            if (npscal_gas == 0) {
+                throw shambase::make_except_with_loc<std::runtime_error>(
+                    "Gas passive scalar is on with npscal_gas == 0");
+            }
+            return true;
+        }
```

Comment:
```suggestion
        inline bool is_gas_passive_scalar_on() {
            return npscal_gas == 0;
        }
```
Here i assume you want to check if you have passive scalars, so you probably don't want to throw an exception?

Assume that raising an exception = to a crash. In general I use them to verify that the state is correct.